runtime/meta.c

```diff
@@ -231,7 +231,7 @@ CAMLprim value caml_invoke_traced_function(value codeptr, value env, value arg)
   Caml_state->extern_sp -= 4;
   nsp = Caml_state->extern_sp;
   for (i = 0; i < 7; i++) nsp[i] = osp[i];
-  nsp[7] = codeptr;
+  nsp[7] = (value) Nativeint_val(codeptr);
```

Comment:
This cast is a bit odd because we know that the data is in fact *not* a valid OCaml value. If we wanted to be more precise about what is going on here, we would type the stack with a union type of either a value or raw data.

Reply:
It is a valid value today, in naked-pointer mode.  But, yes, the stack of the bytecode interpreter is an "interesting" mix of well-formed values, code pointers, and pointers within the stack.  One of my next PRs will deal with that.